### DISCUSSION TITLE
What makes this package mostly undetectable?

### DISCUSSION THREAD

**Discussion by ttlns** [kaliiiiiiiiii/Selenium-Driverless#86] [General]
Basically, frameworks like Selenium, puppeteer or playwright have leaks, as they don't focus on undetectability.
Internally, they all use the [chrome-developer-protocol](https://vanilla.aslushnikov.com/) (**CDP**), which controlls any Chromium based browser in the end.

### So what makes this package different?
1.  It avoids the usage of CDP-commands like `Runtime.enable`, which are easily detectable. 
2.  It avoids executing javascript outside of **isolated worlds** (execution-contexts). In isolated worlds, only the DOM (=>html) is shared. (see `unique_context=True` argument at `target.execute_script`)
3.  Itugenerally **focuses on undetectability**. Most other open-source package don't focus on that, as they're only meant for testing.

**Comment by thecodedokter** [comment]
You are awesome. Really....this is awesome. Great job!!!

**Reply by bfwg** [reply]
What a legend.
